buildSrc/src/main/kotlin/io/spine/internal/gradle/VersionWriter.kt

```diff
@@ -126,15 +126,12 @@ class VersionWriter : Plugin<Project> {
         val task = register("writeVersions", WriteVersions::class.java) {
             versionsFileLocation.convention(project.layout.buildDirectory.dir(name))
             includeOwnVersion()
-            @Suppress("SimpleRedundantLet")
-                // Decrease the number of null-safe calls (`?.`).
-            extensions.findByType<JavaPluginExtension>()?.let {
-                it.sourceSets
-                    .getByName("main")
+            project.extensions.findByType<JavaPluginExtension>()!!.apply {
```

Comment:
As discussed, let's extract this as an extension to `Project` (see https://github.com/SpineEventEngine/config/pull/256/files#diff-6fc4e249496a8134f46be7e5e2ccd80e5f4afaf68124f72fd4414341c8856b75R150).